### PR TITLE
[BD-18] Switch xqueue_watcher from nose to pytest

### PR DESCRIPTION
**Pull request by tehreem-sadat**
**Description:**

- Replace nose testing framework with pytest.

**Packages removed**
nose

**New packages**
pytest-cov

### CODE REVIEW

**Comment by edx-webhook** [comment]
Thanks for the pull request, @tehreem-sadat! I've created [OSPR-4786](https://openedx.atlassian.net/browse/OSPR-4786) to keep track of it in JIRA, where we prioritize reviews.  Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).

Please let us know once your PR is ready for our review and all tests are green.

File: Makefile [hunk 1]
```diff
@@ -2,11 +2,11 @@ NODE_BIN=./node_modules/.bin
 
 help:
 	@echo '                                                                                     		'
-	@echo 'Makefile for the xqueue-watcher                                                   		'
+	@echo 'Makefile for the xqueue-watcher                                                   			'
```
Comment (feanil) [inline Makefile:5]: Please fix the extra spacing.  Perhaps some tabs got converted to spaces or vise-versa.

File: Makefile [hunk 2]
```diff
@@ -2,11 +2,11 @@ NODE_BIN=./node_modules/.bin
...
 	@echo '                                                                                     		'
 	@echo 'Usage:                                                                               		'
 	@echo '    make requirements                 install requirements for local development     		'
-	@echo '    make test                         install requirements for local development     		'
+	@echo '    make test                         run python unit-tests     									'
```
Comment (feanil) [inline Makefile:9]: nice, please fix the spacing.

**Comment by tehreem-sadat** [comment]
@feanil please review it again.

**Pull request merged**